app/post.go

```diff
@@ -139,6 +136,9 @@ func handlePostEvents(post *model.Post, teamId string, triggerWebhooks bool) *mo
 		channel = result.Data.(*model.Channel)
 	}
 
+	InvalidateCacheForChannel(channel)
```

Comment:
It wasn't in CS's original PR but maybe it needs it? https://github.com/mattermost/platform/pull/5210/files